yaptide/persistence/models.py

```diff
@@ -69,22 +75,150 @@ class SimType(Enum):
+                                           doc="Simulation job ID (foreign key)")
+
+    # we encode task_id as string, because that is the convention in Celery queue and in SLURM
+    # currently for Celery one simulation is one job, but in future we might want to split it into multiple tasks
+    # example celery job_ids are: 1a2b3c4d-5e6f-7g8h-9i0j-1k2l3m4n5o6p
+    # example SLURM job_ids are: 12345678_12
+    task_id: Column[str] = db.Column(db.String, nullable=False, unique=True, doc="Task ID")
+    requested_primaries: Column[int] = db.Column(db.Integer,
+                                                 nullable=False,
+                                                 default=0,
+                                                 doc="Requested number of primaries")
+    simulated_primaries: Column[int] = db.Column(db.Integer,
+                                                 nullable=False,
+                                                 default=0,
+                                                 doc="Simulated number of primaries")
+    task_state: Column[str] = db.Column(db.String,
+                                        nullable=False,
+                                        default='PENDING',
+                                        doc="Task state (i.e. 'pending', 'running', 'completed', 'failed')")
+    estimated_time: Column[int] = db.Column(db.Integer, nullable=True, doc="Estimated time in seconds")
+    start_time: Column[datetime] = db.Column(db.DateTime(timezone=True), default=now(), doc="Task start time")
+    end_time: Column[datetime] = db.Column(db.DateTime(timezone=True), nullable=True, doc="Task end time")
+    last_update_time: Column[datetime] = db.Column(
+        db.DateTime(timezone=True),
+        default=now(),
+        doc="Task last update time")
+
+    def update_state(self, update_dict: dict):
+        """
+        Updating database is more costly than a simple query.
+        Therefore we check first if update is needed and
+        perform it only for such fields which exists and which have updated values.
+        """
+        if "requested_primaries" in update_dict and self.requested_primaries != update_dict["requested_primaries"]:
+            self.requested_primaries = update_dict["requested_primaries"]
+        if "simulated_primaries" in update_dict and self.simulated_primaries != update_dict["simulated_primaries"]:
+            self.simulated_primaries = update_dict["simulated_primaries"]
+        if "task_state" in update_dict and self.task_state != update_dict["task_state"]:
+            self.task_state = update_dict["task_state"]
+        # Here we have a special case, `estimated_time` cannot be set when `end_time` is set - it is meaningless
+        have_estim_time = "estimated_time" in update_dict and self.estimated_time != update_dict["estimated_time"]
+        end_time_not_set = self.end_time is None
+        if have_estim_time and end_time_not_set:
```

Comment:
@grzanka is it more readable with variables than well formatted single if statement?
```suggestion
        if ("estimated_time" in update_dict
            and self.estimated_time != update_dict["estimated_time"]
            and self.end_time is None
        )
```

Reply:
Yes, variable names bring more meaning and are much easier to debug.
With variables debugging is much easier (you can add watcher in IDE, print it on the screen etc).
Moreover deep source will not complain about line being too long.
Generally speaking better practice is to have meaningful short functions or variable than complex boolean logic.